http/src/main/java/io/crate/protocols/http/SslContextPipelineRegistryBinder.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.http;
+
+import io.crate.plugin.PipelineRegistry;
+import io.crate.protocols.ssl.SslContextProvider;
+import org.elasticsearch.common.component.AbstractLifecycleComponent;
+import org.elasticsearch.common.inject.Inject;
+
+public class SslContextPipelineRegistryBinder extends AbstractLifecycleComponent {
+
+    private final PipelineRegistry pipelineRegistry;
+    private final SslContextProvider sslContextProvider;
+
+    @Inject
+    public SslContextPipelineRegistryBinder(PipelineRegistry pipelineRegistry,
+                                            SslContextProvider sslContextProvider) {
+        this.pipelineRegistry = pipelineRegistry;
+        this.sslContextProvider = sslContextProvider;
+    }
+
+    @Override
+    protected void doStart() {
+        pipelineRegistry.setSslContextProvider(sslContextProvider);
```

Comment:
Maybe we could to this in `RestSQLAction` which is already responsible for setting up the sql http handler